core/src/main/java/io/fabric8/maven/core/access/ClusterConfiguration.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2018 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.access;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.maven.core.util.kubernetes.KubernetesHelper;
+import org.apache.commons.lang3.StringUtils;
+
+public class ClusterConfiguration {
+
+    private String masterUrl;
+    private String apiVersion;
+    private String namespace;
+    private String caCertFile;
+    private String caCertData;
+    private String clientCertFile;
+    private String clientCertData;
+    private String clientKeyFile;
+    private String clientKeyData;
+    private String clientKeyAlgo;
+    private String clientKeyPassphrase;
+    private String trustStoreFile;
+    private String trustStorePassphrase;
+    private String keyStoreFile;
+    private String keyStorePassphrase;
+
+    public ClusterConfiguration() {
+    }
+
```

Comment:
I would also add an `fromProps(Properties)` to a potential builder method to extract the values from system/maven properties. This would have the additional advantage that the property names are only used in such a method and would be duplicated over multiple mojos.